src/Block/FeatureMediaBlockService.php

```diff
@@ -13,22 +13,65 @@
 
 namespace Sonata\MediaBundle\Block;
 
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Form\Type\ModelListType;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Block\Service\AbstractBlockService;
+use Sonata\BlockBundle\Block\Service\EditableBlockService;
 use Sonata\BlockBundle\Form\Mapper\FormMapper;
 use Sonata\BlockBundle\Meta\Metadata;
 use Sonata\BlockBundle\Meta\MetadataInterface;
 use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\Doctrine\Model\ManagerInterface;
 use Sonata\Form\Type\ImmutableArrayType;
 use Sonata\Form\Validator\ErrorElement;
+use Sonata\MediaBundle\Model\MediaInterface;
+use Sonata\MediaBundle\Provider\Pool;
 use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
 use Symfony\Component\Form\Extension\Core\Type\TextareaType;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\OptionsResolver\OptionsResolver;
+use Twig\Environment;
 
 /**
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
-final class FeatureMediaBlockService extends MediaBlockService
+final class FeatureMediaBlockService extends AbstractBlockService implements EditableBlockService
 {
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var AdminInterface<MediaInterface>
+     */
+    private $mediaAdmin;
+
+    /**
+     * @var ManagerInterface<MediaInterface>
+     */
+    private $mediaManager;
+
+    /**
+     * @param AdminInterface<MediaInterface>   $mediaAdmin
+     * @param ManagerInterface<MediaInterface> $mediaManager
+     */
+    public function __construct(
+        Environment $twig,
+        Pool $pool,
+        AdminInterface $mediaAdmin,
+        ManagerInterface $mediaManager
```

Comment:
You can use `MediaManagerInterface` here

Reply:
Replaced all usages of ManagerInterface by the correct interface (Media or Gallery).